### PR TITLE
Add a Server type

### PR DESCRIPTION
**Pull request by empijei**
This allows to construct a ready-to use and safe server from a Mux and an address

### CODE REVIEW

File: safehttp/server.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"crypto/tls"
+	"net"
+	"net/http"
+	"time"
+)
+
+// Server is a safe wrapper for a standard HTTP server.
+type Server struct {
+	srv *http.Server
+}
```
Comment (kele) [inline safehttp/server.go:28]: The `http.Server` zero value is valid. Is there a way we could preserve this in our package?

File: safehttp/server.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	}
+	if s.MaxHeaderBytes != 0 {
+		srv.MaxHeaderBytes = 10 * 1024
+	}
+	if s.TLSConfig != nil {
+		cfg := s.TLSConfig.Clone()
+		cfg.MinVersion = tls.VersionTLS12
+		cfg.PreferServerCipherSuites = true
+		srv.TLSConfig = cfg
+	}
```
Comment (kele) [inline safehttp/server.go:118]: Please document these overrides.

File: safehttp/server.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	if s.WriteTimeout != 0 {
+		srv.WriteTimeout = s.WriteTimeout
+	}
+	if s.IdleTimeout != 0 {
+		srv.IdleTimeout = s.IdleTimeout
+	}
+	if s.MaxHeaderBytes != 0 {
+		srv.MaxHeaderBytes = 10 * 1024
+	}
```
Comment (kele) [inline safehttp/server.go:112]: Why is this set this way?

**Review by kele — CHANGES_REQUESTED** [review]

File: safehttp/server.go (hunk 3 above)
Comment (empijei) [reply safehttp/server.go:112]: The default is a very very high and unreasonable value. 10K should be enough as a default. According to [this](http://dev.chromium.org/spdy/spdy-whitepaper) the average is about 2K, so I decided to leave a bit more room for it.

Comment (kele) [reply safehttp/server.go:112]: But, if I set `s.MaxHeaderBytes` to `123`, why does it get overridden to `10*1024`?

**Review by kele — APPROVED** [review]

**Pull request merged**